validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinator.java

```diff
@@ -212,6 +220,7 @@ public void onAttestationEvent(BroadcastAttestationEvent event) throws IllegalAr
       asyncProduceAttestations(
           attesterInformations, headState, getGenericAttestationData(headState, headBlock));
 
+      latestBroadcastAttestationEventSlot = slot;
```

Comment:
I suspect we should do this in the opposite order.  Otherwise we risk publishing attestations and then failing unexpectedly before we update the `latestBroadcastAttestationEventSlot` which opens us to being slashed.

As mentioned above (sorry for scattering this across multiple comments) - let's pull this out of this PR and do it separately.

Reply:
I'd say let's keep this in this PR to be fully done with the v0.10.1 change and if we want to make it more robust we can make a ticket and come back to it later on.

Reply:
Unless this seems like it will definitely fail in production, after I put these lines in the opposite order, I see no problem merging this in for now.